### PR TITLE
Integrating FFHT into Quspin-Rust

### PR DESCRIPTION
**Pull request by sourav-condmat**
This PR integrates the [FFHT](https://github.com/FALCONN-LIB/FFHT) C library into QuSpin, exposing a Fast Hadamard Transform (FHT) implementation across three layers: C → Rust → Python.

**Architecture**

[`quspin-ffht`](crates/quspin-ffht) (new Rust crate): a safe Rust wrapper around the upstream FFHT C sources (fht_scalar.c, fht_sse.c, fht_avx.c, fast_copy.c), compiled via a custom build.rs. The crate exposes fht_f32, fht_f64, fht_f32_oop, and fht_f64_oop, with runtime CPU-feature dispatch (AVX2+FMA → SSE2 → scalar) selected automatically via is_x86_feature_detected!. This crate is fully self-contained and can be used independently of QuSpin or the Python bindings.

[`quspin-core`](crates/quspin-core): re-exports quspin-ffht as quspin_core::ffht.

[`quspin-py`](crates/quspin-py): exposes the four typed functions (fht_f32, fht_f64, fht_f32_oop, fht_f64_oop) as #[pyfunction]s registered on the _rs extension module, with input validation (C-contiguity, power-of-two length) translated into ValueError/TypeError before calling into Rust.

[`quspin_rs.ffht`](python/quspin_rs/ffht.py): a thin dispatcher exposing a single ffht(arr, inplace=False) entry point that automatically routes to the correct f32/f64 implementation based on arr.dtype, and to the in-place or out-of-place variant based on the inplace flag. Re-exported at the package root as quspin_rs.ffht.

**Build system**

A portable [`build.rs`](crates/quspin-ffht/build.rs) compiles each SIMD variant (scalar / SSE2 / AVX2+FMA) of the upstream C sources into separate static libraries. The upstream C files all define functions under the same names (e.g. fht_float, fht_double), since a traditional C build only ever compiles one variant for a given target — but Rust's runtime dispatch (via is_x86_feature_detected!) requires all three variants to coexist in a single binary, which means each variant's symbols must be distinct at link time. [`build.rs`](crates/quspin-ffht/build.rs) resolves this by applying a build-time symbol-renaming pass to the in-memory source before compiling each variant, avoiding collisions without modifying the vendored upstream sources. SSE2/AVX2 variants are built only on x86/x86_64 with non-MSVC compilers (the upstream sources use GNU inline assembly unsupported by cl.exe); on ARM or MSVC, the build falls back to the scalar implementation transparently.

**Diagnostics**

When no SIMD-accelerated implementation is available for the running CPU/build (e.g. ARM, or an x86 CPU without AVX2/SSE2, or an MSVC build), [`quspin-ffht`](crates/quspin-ffht) emits a one-time warning to stderr on first use, informing the user that the scalar fallback is active and performance will be reduced. This is implemented at the lowest dispatch layer, so it surfaces transparently regardless of whether FFHT is invoked from Rust or Python.

**Testing**

Rust integration tests in [`quspin-ffht`](crates/quspin-ffht/tests/ffht.rs) cover correctness (involution property, known reference values cross-checked against the Python suite), in-place/out-of-place agreement, edge cases (n=1), and contract violations (#[should_panic] on non-power-of-two input).

Python tests [`python`](python/tests/ffht/test_ffht.py) covers the same correctness properties at both the low-level _rs.fht_* and high-level quspin_rs.ffht layers, plus input-validation behavior specific to the Python boundary (ValueError on non-contiguous/non-power-of-two arrays, TypeError on dtype mismatch).

### CODE REVIEW

File: crates/quspin-ffht/src/ffht.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+/// is used, so users know SSE2/AVX2 weren't available (either not
+/// compiled in, or not supported by the running CPU) and performance
+/// will be slower than the SIMD path.
+#[inline]
+fn warn_scalar_fallback_once() {
+    SCALAR_FALLBACK_WARNING.call_once(|| {
+        eprintln!(
+            "quspin-ffht: no fast (SSE2/AVX2) implementation available for this \
+             CPU/build; falling back to the scalar implementation. Performance \
+             will be reduced. (This message is shown only once.)"
```
Comment (weinbe58) [inline crates/quspin-ffht/src/ffht.rs]:
```suggestion
             will be reduced."
```

nit: don't need the extra comment on the message.

File: crates/quspin-py/src/ffht.rs
Comment (weinbe58) [inline crates/quspin-py/src/ffht.rs:1]: Instead of adding dynamic dispatch in python file please put the dispatch inside the rust wrapper here. 

here is an example in how to unpack a python object into a vector: https://github.com/QuSpin/QuSpin-rust/blob/a0ad6c9fe2e8063208f9ba1c6677150c993bb554/crates/quspin-py/src/expm.rs#L224

File: python/quspin_rs/ffht.py
Comment (weinbe58) [inline python/quspin_rs/ffht.py:1]: delete this after the rust dispatch is implemented.

**Review by weinbe58 — CHANGES_REQUESTED** [review]
Looks good I just want to put the dispatch into the rust code to remove the extra python module.

File: crates/quspin-ffht/src/ffht.rs (hunk 1 above)
Comment (sourav-condmat) [reply crates/quspin-ffht/src/ffht.rs]: Removed that extra comment in the bracket.

File: crates/quspin-py/src/ffht.rs
Comment (sourav-condmat) [reply crates/quspin-py/src/ffht.rs:1]: Done, ffht.rs now has one public wrapper function which dispatches to the appropriate function depending on the array type.

File: python/quspin_rs/ffht.py
Comment (sourav-condmat) [reply python/quspin_rs/ffht.py:1]: Done, dispatch now directly from [`quspin-py`](crates/quspin-py/src/ffht.rs)

**Review by weinbe58 — APPROVED** [review]
Great work! Thanks!
